adr/021-validation-engine.md

```diff
@@ -0,0 +1,29 @@
+# 21. Validation Engine
+
+Date: 2024-06-11
+
+## Decision
+
+We will implement a Validation Engine using the [Rules Engine Pattern](https://deviq.com/design-patterns/rules-engine-pattern) to validate incoming FHIR messages based on given conditions. Each validation will be represented as a rule, specifying conditions that must be met and the corresponding failure message if validation fails. Initially, rules will be stored in a JSON file, with considerations on transitioning to a database for scalability.
```

Comment:
For the validation and transformation engines, it might be useful to call out what RS does vs what we do in TI and how those are different

Reply:
I think what RS would be outside of the scope of these ADRs as they are not about the transformation or validation themselves, but about the framework that implements them. The engines are meant to be generic and the FHIR that gets validated or transformed could be coming from anywhere (in fact, I think in the future could be a good idea to spin off the engines into their own services)